app.py

```diff
@@ -12,6 +12,12 @@
 
 @app.route('/forecast/single-day', methods=['POST'])
 def get_weather():
+    try:
+        data = request.get_json()
+        request_model = ForecastRequestModel(**data)
+        datetime.strptime(request_model.date, '%Y-%m-%d')  # Date validation
+    except (ValidationError, ValueError) as e:
+        return jsonify({"error": str(e)}), 400
     try: 

```

Comment:
You added the new try/catch in front of the old one. The new try/catch is causing the errors when not sending a date. You need to add the `if 'date' not in data or not data['date']:
data['date'] = datetime.now().strftime('%Y-%m-%d')` back before line 17 then put the the validation on line 18 in an else statement if the date is not empty